commit 69db7df9942b13a502534baba4af6932946d45ed
Author: Phil Elwell <phil@raspberrypi.org>
Date:   Tue Feb 23 17:26:48 2016 +0000

    amba_pl011: Don't use DT aliases for numbering
    
    The pl011 driver looks for DT aliases of the form "serial<n>",
    and if found uses <n> as the device ID. This can cause
    /dev/ttyAMA0 to become /dev/ttyAMA1, which is confusing if the
    other serial port is provided by the 8250 driver which doesn't
    use the same logic.

diff --git a/drivers/tty/serial/amba-pl011.c b/drivers/tty/serial/amba-pl011.c
index f2503d862f3a..a85e81245004 100644
--- a/drivers/tty/serial/amba-pl011.c
+++ b/drivers/tty/serial/amba-pl011.c
@@ -2582,7 +2582,12 @@ static int pl011_setup_port(struct device *dev, struct uart_amba_port *uap,
 	if (IS_ERR(base))
 		return PTR_ERR(base);
 
+	/* Don't use DT serial<n> aliases - it causes the device to
+	   be renumbered to ttyAMA1 if it is the second serial port in the
+	   system, even though the other one is ttyS0. The 8250 driver
+	   doesn't use this logic, so always remains ttyS0.
 	index = pl011_probe_dt_alias(index, dev);
+	*/
 
 	uap->old_cr = 0;
 	uap->port.dev = dev;
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1091_69db7df9942b13a502534baba4af6932946d45ed.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1091_69db7df9942b13a502534baba4af6932946d45ed.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
